art/attacks/evasion/graphite/graphite_blackbox.py

```diff
@@ -0,0 +1,896 @@
+from art.attacks.evasion.graphite.utils import (
+    convert_to_network,
+    get_transform_params,
+    add_noise,
+    get_transformed_images,
+    run_predictions,
+    score_fn,
+)
+
+if TYPE_CHECKING:
+    from art.utils import CLASSIFIER_NEURALNETWORK_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class GRAPHITEBlackbox(EvasionAttack):
+    """
+    Implementation of the hard-label GRAPHITE attack from Feng et al. (2022). This is a physical, black-box attack
+    that only requires final class prediction and generates robust physical perturbations that can be applied as
+    stickers.
+
+    | Paper link: https://arxiv.org/abs/2002.07088
+    | Original github link: https://github.com/ryan-feng/GRAPHITE
+    """
+
+    attack_params = EvasionAttack.attack_params + [
+        "noise_size",
+        "net_size",
+        "heat_patch_size",
+        "heat_patch_stride",
+        "heatmap_mode",
+        "tr_lo",
+        "tr_hi",
+        "num_xforms_mask",
+        "max_mask_size",
+        "rotation_range",
+        "beta",
+        "eta",
+        "num_xforms_boost",
+        "num_boost_queries",
+        "dist_range",
+        "gamma_range",
+        "crop_percent_range",
+        "off_x_range",
+        "off_y_range",
+        "blur_kernels",
+        "batch_size",
+    ]
+    _estimator_requirements = (BaseEstimator, ClassifierMixin)
+
+    def __init__(
+        self,
+        classifier: "CLASSIFIER_NEURALNETWORK_TYPE",
+        noise_size: Tuple[int, int],
+        net_size: Tuple[int, int],
+        heat_patch_size: Tuple[int, int] = (4, 4),
+        heat_patch_stride: Tuple[int, int] = (1, 1),
+        heatmap_mode: str = "Target",
+        tr_lo: float = 0.65,
+        tr_hi: float = 0.85,
+        num_xforms_mask: int = 100,
+        max_mask_size: int = -1,
+        beta: float = 1.0,
+        eta: float = 500,
+        num_xforms_boost: int = 100,
+        num_boost_queries: int = 20000,
+        rotation_range: Tuple[float, float] = (-30.0, 30.0),
+        dist_range: Tuple[float, float] = (0.0, 0.0),
+        gamma_range: Tuple[float, float] = (1.0, 2.0),
+        crop_percent_range: Tuple[float, float] = (-0.03125, 0.03125),
+        off_x_range: Tuple[float, float] = (-0.03125, 0.03125),
+        off_y_range: Tuple[float, float] = (-0.03125, 0.03125),
+        blur_kernels: Union[Tuple[int, int], List[int]] = (0, 3),
+        batch_size: int = 64,
+    ) -> None:
+        """
+        Create a GRAPHITEBlackbox attack instance.
+
+        :param classifier: A trained classifier.
+        :param noise_size: The resolution to generate perturbations in (w, h).
+        :param net_size: The resolution to resize images to before feeding to the model in (w, h).
+        :param heat_patch_size: The size of the heatmap patches in (w, h).
+        :param heat_patch_stride: The stride of the heatmap patching in (w, h).
+        :param heatmap_mode: The mode of heatmap in ['Target', 'Random'].
```

Comment:
This is a bit vague. I assume this refers to the heatmap generation method. The values can be ['Target', 'Random']. A minor suggestion would be to not make this variable case sensitive.